lib/mongoid/extensions/binary.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Mongoid
+  module Extensions
+    module BSON
+      module Binary
```

Comment:
The file should also be nested under `bson` subdir.

Reply:
After looking at what we've been doing I think I should remove the BSON module since we already have `Mongoid::Extensions::ObjectId`, unless you want me to also move that module, which would be backwards breaking.